src/datalinker/dependencies/hips.py

```diff
@@ -3,22 +3,134 @@
 import re
 from typing import Annotated
 
-from fastapi import Depends
+from fastapi import Depends, HTTPException
 from httpx import AsyncClient
+from pydantic import HttpUrl
 from safir.dependencies.http_client import http_client_dependency
 from safir.dependencies.logger import logger_dependency
 from structlog.stdlib import BoundLogger
 
-from ..config import config
+from ..config import Config
 from ..constants import HIPS_DATASETS
+from .config import config_dependency
 
 __all__ = [
+    "DatasetHiPSListDependency",
     "HiPSListDependency",
+    "dataset_hips_list_dependency",
     "hips_list_dependency",
 ]
 
 
-class HiPSListDependency:
+class BaseHiPSListDependency:
+    """Base class with common HiPS list functionality."""
+
+    async def _fetch_and_process_properties(
+        self,
+        base_url: HttpUrl,
```

Comment:
Over three parameters, so I would add `*,` here to force use of key/value pair parameters instead of positional parameters.